src/cljs/org/broadinstitute/firecloud_ui/common/table_utils.cljs

```diff
@@ -24,65 +24,73 @@
+                [:span {:style {:whiteSpace "nowrap"}}
                  (map (fn [n]
                         (let [selected? (= n current-page)]
-                          [:div {:style {:paddingTop 5 :display "inline-block" :width 29 :height 24
+                          [:div {:style {:textAlign "center"
+                                         :paddingTop 5 :display "inline-block" :width 29 :height 24
                                          :backgroundColor (when selected? (:button-blue style/colors))
                                          :color (if selected? "white" (:button-blue style/colors))
                                          :borderRadius (when selected? "100%")
                                          :cursor (when-not selected? "pointer")}
                                  :onClick (when-not selected? #((:on-change props) (assoc-in pagination-params [:current-page] n)))}
                            n]))
                    (create-page-range current-page num-pages))]
-                [:div {:style {:display "inline-block" :padding "0.55em 0.9em"
+                [:div {:style {:display "inline-flex" :alignItems "baseline"
+                               :padding "0em 0.9em"
                                :color (if allow-next
                                         (:button-blue style/colors)
                                         (:border-gray style/colors))
                                :cursor (when allow-next "pointer")}
                        :onClick (when allow-next #((:on-change props) (update-in pagination-params [:current-page] inc)))}
-                 [:span {:style {:paddingRight "1em"}} "Next"]
-                 (icons/font-icon {:style {:fontSize "70%"}} :angle-right)])
+                 "Next"
+                 (icons/icon {:style {:alignSelf "center" :margin "-0.5em 0"}} :angle-right)])
               rows-component
-              [:div {:style (merge {:textAlign "right"} layout-style)}
+              [:div {:style {:display "inline-flex" :alignItems "baseline"}}
                "Display"
                (style/create-select
                  {:style {:width 60 :margin "0em 1em"}
                   :onChange #((:on-change props) {:rows-per-page (nth rows-per-page-options (-> % .-target .-value js/parseInt))
                                                   :current-page 1})}
                  rows-per-page-options)
                "rows per page"]]
-          [:div {:style {:fontSize 13 :lineHeight 1.5 :padding "0px 48px"}}
-           view-component page-component rows-component
-           (common/clear-both)])]))})
+          [:div {:style {:fontSize 13 :lineHeight 1.5 :padding "0 48px"
+                         :display "flex" :flexDirection (if narrow? "column" "row")
+                         :alignItems (if narrow? "center" "baseline")}}
+           (container view-component "left")
+           (container page-component "center")
+           (container rows-component "right")])]))})
```

Comment:
The layout of the paginator went to shit surprisingly hard with the icon swap.  So I had to completely redo the layout.